src/fmu/dataio/_metadata.py

```diff
@@ -24,6 +38,38 @@
 logger: Final = null_logger(__name__)
 
 
+class JsonSchemaMetadata(BaseModel):
```

Comment:
Thought for a separate PR, could this be removed from here and used in the `_model.root` instead? 🙂 

Right now we have a mismatch between the `$schema` defined in the main schema and the "exported". That would take us one step closer to removing the need for a separate model for the exported metadata, I look forward to the day we can align them fully!

Reply:
I think it can't actually, this is a bit confusing:

- The 'Root' json schema is validated by $schema = https://json-schema.org/draft/2020-12/schema
- Metadata we export is validated against $schema = https://main-fmu-schemas-prod.radix.equinor.com/schemas/0.8.0/fmu_results.json

It's confusing because _our schema_ is getting validated by a 'worldwide' schema (validating that it's a valid schema), while _our metadata_ is validated by _our schema_

Reply:
annoying :weary: .. and confusing yes